app-ingest/src/main/scala/com/azavea/rf/JobDefinition.scala

```diff
@@ -0,0 +1,55 @@
+package com.azavea.rf
+
+import spray.json._
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+
+import geotrellis.vector.Extent
+import geotrellis.vector.io._
+
+import java.net.URL
+
+/** Each TileSource, which sits the bottom of any JobDefinition (see below) is made of tiff urls and a target extent */
+case class TileSource(urls: Array[String], extent: Extent)
```

Comment:
Basically we have to be able to remap, collect or break out bands from multiple files. Hopefully this makes sense:

```json
// scene: Collection of images which share metadata, captured at the same time, possibly chunked into multiple files.
// uri: Can be a prefix, a full file name, or wild-card
// extent: Lat-Long extent covering a single image or a collection if they are prefixed/wild-carded
//   Ideally extent is optional and can be collected if not given
// pyramid: A GeoTrellis pyramid for a scene built from scene resolution to 1

// Uses cases covered:
// 1. Many files cover a single scene, single band (ex: NLCD)
// 2. Many files cover a single scene, multiple bands
// 3. Many files cover a single scene, band re-ordered
// 4. Many files cover a single scene, a band per file (ex: Landsat)
// 5. Many files cover a single scene, band sub-select (ex: none that I know of)

// Use Case 1
{
    "output": "s3://raster-foundry/user-bucket/pyramid-prefix",
    "scene": [
        {
            "uri": ["s3://user-bucket/scene-prefix"],
            "extent": [0, 0, 10, 10]
        }
    ]
}


// Use Case 2/3
{
    "output": "s3://raster-foundry/user-bucket/pyramid-prefix",
    "scene": [
        {
            "uri": ["s3://user-bucket/scene-prefix"],
            "extent": [0, 0, 10, 10],
            "bands": [2,1,6] // chance re-order bands
        }
    ]
}

// Use Case 4
{
    "output": "s3://raster-foundry/user-bucket/pyramid-prefix",
    "scene": [
        [
            { // Band 0
                "uri": ["s3://user-bucket/scene-prefix/LC8_B3.tif"],
                "extent": [0, 0, 10, 10]
            },
            { // Band 1
                "uri": ["s3://user-bucket/scene-prefix/LC8_B2.tif"],
                "extent": [0, 0, 10, 10]
            }
        ]
    ]
}

// Use Case 5
{
    "output": "s3://raster-foundry/user-bucket/pyramid-prefix",
    "scene": [
        [
            { // Band 0
                "uri": ["s3://user-bucket/scene-prefix/LC8_B3.tif"],
                "extent": [0, 0, 10, 10],
                "band": 1
            },
            { // Band 1
                "uri": ["s3://user-bucket/scene-prefix/LC8_B2.tif"],
                "extent": [0, 0, 10, 10],
                "band": 1
            }
        ]
    ]
}
// Job, collection of scenes
{
    "name": "My Job",
    "id": "uuid-000-0000",
    "pyramids": [ { // pyramid objects }]
}
```

Doing wild-cards would obviously require some code, but it would be well worth it for the expressiveness.